Services/AuthenticationService.cs

```diff
@@ -5,133 +5,160 @@
         {
-            var user = await _userRepository.GetAll().Where(a => a.Id == _httpContextService.GetCurrentUserID()).SingleAsync();
+            UserID = user.Id.ToString(),
+            ExpireDate = DateTime.UtcNow.Add(TimeSpan.FromMinutes(5)),
+            Role = user.Role
+        });
 
-            //Hashing the password for security
-            string hashedPassword = BCrypt.Net.BCrypt.HashPassword(request.Password);
+        await _emailSender.SendEmailAsync(new SendEmailRequest
+        {
+            ReceiverMail = request.Email,
+            MailType = MailType.ForgotPasswordMail
+        });
 
-            user.RoleId = 2;
-            user.Password = hashedPassword;
+        return await generatedToken;
+    }
 
-            await _userRepository.UpdateAsync(user);
+    public async Task<UserRegisterResponse> RegisterUserAsync(UserRegisterRequest request)
+    {
+        User newUser = new User()
+        {
+            FirstName = request.FirstName,
+            LastName = request.LastName,
+            PhoneNumber = request.PhoneNumber,
+            BirthDate = request.BirthDate,
+            Email = request.Email,
+            Gender = request.Gender,
+            BirthCountry = request.PhoneNumber,
+            BirthCity = request.Email,
+            CurrentCountry = request.CurrentCountry,
+            CurrentCity = request.CurrentCity,
+            Address = request.Address,
+            RoleId = 3,
+        };
+
+        var user = await _userRepository.AddAsync(newUser);
+        user = await _userRepository.GetAll().Where(u => u.Id == user.Id).Include(u => u.Role).SingleAsync(); //aşağıda role kullandığım için roleid ile eşlemiyordu.
+
+
+        foreach (var interestId in request.InterestIdList)
+        {
+            var interest = await _interestRepository.GetByIdAsync(interestId);
+            await _userRepository.AddInterest(user.Id, interest);
         }
+
+        //Create token for password
+        var generatedToken = await _tokenService.GenerateToken(new GenerateTokenRequest
+        {
+            UserID = user.Id.ToString(),
+            Role = user.Role,
+            ExpireDate = DateTime.UtcNow.Add(TimeSpan.FromMinutes(5))
+        });
+
+        await _emailSender.SendEmailAsync(new SendEmailRequest
+        {
+            ReceiverMail = newUser.Email,
+            MailType = MailType.SetPasswordMail
+        });
+
+        return new UserRegisterResponse()
+        {
+            isSuccess = true,
+            Token = generatedToken.Token,
+            TokenExpireDate = generatedToken.TokenExpireDate
+        };
+    }
+
+    public async Task SetPasswordAsync(SetPasswordRequest request)
+    {
+        var user = await _userRepository.GetAll().Where(a => a.Id == _httpContextService.GetCurrentUserID()).SingleAsync();
+
+        //Hashing the password for security
+        string hashedPassword = BCrypt.Net.BCrypt.HashPassword(request.Password);
```

Comment:
we can see that you hash the password for the security. Remove unnecesery comment